hxjo/fastapi-inertia
# fix: add support for vite react plugin. Include react example

**What does this PR do?**
This PR supports adding a tag for `inertia_react_refresh` in the html template to enable this library work with react + vite in development

**Why is this PR needed**
Running a project using react + vite, with inertia and fastapi inertia will result in the error `vite-plugin-react can't detect preamble. Something is wrong.`. Read more about this error[here](https://github.com/vitejs/vite/issues/1984)

**How can this be manually tested**
- Scaffold a project using the [vite + react](https://vitejs.dev/guide/#scaffolding-your-first-vite-project) setup
- Setup inertia in your react project
- Setup a fast api project using fast api inertia
- In your `index.html` template, add `inertia_react_refresh` before `inertia_head`
- Run the fast api project and then run the react project

Inspecting the head of your source file on the browser should return contain script similar to this:
![Screenshot 2024-08-06 at 12 25 44](https://github.com/user-attachments/assets/2bc1bd51-fade-4b03-bfec-6deac8e69002)
